code/game/objects/items/mail_box.dm

```diff
@@ -0,0 +1,119 @@
+/*
+
+	Mailboxes/P.O. Boxes! (idk which one fits more)
+	Contains mail, using it with an empty hand opens a menu for you to pull out your own mail.
+	Having cargo bay access lets you pull out anyone's mail, so it doesn't get trapped in limbo
+
+	made by candycane/etherware
+
+*/
+
+
+/obj/structure/mailbox
+	name = "official postbox"
+	desc = "A small postbox stationed for easy access, containing station letters."
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "mailbox"
+	density = TRUE
+	max_integrity = 1000
+	req_access = list(ACCESS_CARGO)  // can look thru anyones mail
+	var/list/active_slots = list()
+	var/list/spam_mail = list()
+
+/obj/machinery/mailbox/examine(mob/user)
```

Comment:
you do actually have to change every `/obj/machinery/mailbox` to `/obj/structure/mailbox`. Also don't forget the maps

Reply:
oops